decidim-proposals/app/commands/decidim/proposals/create_proposal.rb

```diff
@@ -41,18 +41,27 @@ def call
 
       attr_reader :form, :proposal, :attachment
 
-      def create_proposal
+      def proposal_attributes
+        fields = {}
+
         parsed_title = Decidim::ContentProcessor.parse_with_processor(:hashtag, form.title, current_organization: form.current_organization).rewrite
         parsed_body = Decidim::ContentProcessor.parse_with_processor(:hashtag, form.body, current_organization: form.current_organization).rewrite
 
-        @proposal = Proposal.new(
-          title: parsed_title,
-          body: parsed_body,
-          component: form.component
-        )
-        @proposal.add_coauthor(@current_user, user_group: user_group)
-        @proposal.save!
-        @proposal
+        fields[:title] = parsed_title
+        fields[:body] = parsed_body
+        fields[:component] = form.component
+
+        fields
+      end
+
+      # Prevent PaperTrail from creating a version
+      # in the proposal multi-step creation process
```

Comment:
This docs explain the what/how, can you elaborate on the why?

Reply:
By default, a newly created proposal would have more than one version because the creation process is multi-step:

- Create your proposal
- Compare
- Complete
- Publish your proposal

I understand the desired behaviour is to have only one version to show in the version control feature, so I had to get rid of the others. Instead of creating and deleting the ones I don't wanted (which IS an option), I've chosen to explicitly forbid PaperTrail from creating versions in the creation process and set the final version in the last step.

Reply:
I'm refactoring the code to make it simpler.

Reply:
Thanks very much. I've refactored the code and now it simpler. Sorry for overengineering, this was done a few weeks ago when I didn't understand the code as much.